itl80211/openbsd/net80211/ieee80211_crypto.c

```diff
@@ -384,7 +386,7 @@ ieee80211_kdf(const u_int8_t *key, size_t key_len, const u_int8_t *label,
         HMAC_SHA256_Init(&ctx, key, key_len);
         iter = htole16(i);
         HMAC_SHA256_Update(&ctx, (u_int8_t *)&iter, sizeof iter);
-        HMAC_SHA256_Update(&ctx, label, label_len);
+        HMAC_SHA256_Update(&ctx, label, strlen((const char*)label));
```

Comment:
If this is wrong, what about `ieee80211_prf`?

Reply:
This is one of the most confusing part. I didn't read the original 11w specification. I learnt this from reading the wpa_supplicant.
There are 2 major differences compare between HMAC-SHA1 PRF vs HMAC-SHA256 KDF/PRF.
- When using the `Pairwise key expansion` label, there are differences counting the string length. In [SHA1 PRF](https://w1.fi/cgit/hostap/tree/src/crypto/sha1-prf.c#n36), the label is taken as `Pairwise key expansion\0`. In [SHA256 PRF](https://w1.fi/cgit/hostap/tree/src/crypto/sha256-prf.c#n69), the label is taken as `Pairwise key expansion`. This one-byte difference is causing incorrect key being generated.
- The SHA256 KDF is also taken the number of output bits as part of its data, where the SHA1 PRF doesn't. This make difference since SHA256 KDF is expecting the output PTK to be 48 bytes, the previous code is passing as 64 bytes.